{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "mount_file_id": "17TZ14SlhpIR5bPZCCpjjWAiLXZMNzMPw",
      "authorship_tag": "ABX9TyMgV0jobWrNZj1YfI0AOMtd",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/ryuchallenger/red_project/blob/main/SCA_PC_AL_MC.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "id": "bYtdKNyT3Go-"
      },
      "outputs": [],
      "source": [
        "### library\n",
        "\n",
        "import pandas as pd\n",
        "import numpy as np\n",
        "import re\n",
        "import matplotlib.pyplot as plt"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "### Data\n",
        "\n",
        "PC=pd.read_csv(\"/content/drive/MyDrive/SCA/SCA21_NON47/SCA_rawdata/PC.tsv\", sep=\"\\t\", index_col=\"TRID\")\n",
        "NC=pd.read_csv(\"/content/drive/MyDrive/SCA/SCA21_NON47/SCA_rawdata/NC.tsv\", sep=\"\\t\")\n",
        "STATUS=pd.read_csv(\"/content/drive/MyDrive/SCA/SCA21_NON47/SCA21/PC_AL_all_status_long.tsv\", sep=\"\\t\")\n",
        "STRchive=pd.read_csv(\"/content/drive/MyDrive/SCA/SCA21_NON47/NON47/Genotyping-STRchive-disease-loci-v2.4.2.hg38.TRGT.B2FLLAlV.bed\", sep=\"\\t\", header=None)\n",
        "STRchive.columns=[\"chromosome\", \"start\", \"end\", \"description\"]"
      ],
      "metadata": {
        "id": "oYk1_58k5muH"
      },
      "execution_count": 96,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "### SCA_RED\n",
        "\n",
        "SCA_RED = [\"DRPLA_ATN1\",\n",
        "           \"SCA1_ATXN1\",\n",
        "           \"SCA2_ATXN2\",\n",
        "           \"SCA3_ATXN3\",\n",
        "           \"SCA4_ZFHX3\",\n",
        "           \"SCA6_CACNA1A\",\n",
        "           \"SCA7_ATXN7\",\n",
        "           \"SCA8_ATXN8OS\",\n",
        "           \"SCA10_ATXN10\",\n",
        "           \"SCA12_PPP2R2B\", #less commonly\n",
        "           \"SCA17_TBP\",\n",
        "           \"SCA27B_FGF14\",\n",
        "           \"FRDA_FXN\",\n",
        "           \"CANVAS_RFC1\",\n",
        "           \"SCA31_BEAN1\", #less commonly\n",
        "           \"FXS_FMR1\", #less commonly\n",
        "           \"SCA36_NOP56\", #less commonly\n",
        "           \"SCA37_DAB1\"]\n",
        "\n",
        "OMIM = [\n",
        "    \"The ATN1 gene encodes atrophin-1, a member of a class of evolutionarily conserved transcriptional corepressors involved in nuclear signaling. ATN1 is believed to play a role as a nuclear transcriptional regulator important for brain and other organ system development (summary by Palmer et al., 2019).\",\n",
        "    \"ATXN1 binds RNA, associates with large protein complexes, and interacts with a vast network of proteins. ATXN1 is thought to be involved in transcriptional repression and to regulate Notch (see 190198)- and Capicua (CIC; 612082)-controlled developmental processes (summary by Bergeron et al., 2013).\",\n",
        "    \"Ataxin-2, the protein encoded by the ATXN2 gene, contains a polyglutamine tract, long expansions (greater than 33 repeats) of which result in spinocerebellar ataxia-2 (SCA2; 183090), an autosomal dominant form of olivopontocerebellar atrophy. Intermediate-length expansions (27-33 glutamines) contribute to susceptibility to amyotrophic lateral sclerosis (ALS13; see 183090).\",\n",
        "    \"ATXN3 has deubiquitinase activity and appears to be a component of the ubiquitin proteasome system. It may also have roles in transcriptional regulation and neuroprotection (summary by Haacke et al., 2006).\",\n",
        "    \"ZFHX3 is a transcriptional regulator that interacts with AT motifs in target genes. ZFHX3 is involved in regulation of circadian rhythm (Parsons et al., 2015) and appetite and body weight (Nolan et al., 2023).\",\n",
        "    \"The CACNA1A gene encodes the transmembrane pore-forming subunit of the P/Q-type or CaV2.1 voltage-gated calcium channel (VGCC) (Kordasiewicz et al., 2006). Voltage-dependent Ca(2+) channels not only mediate the entry of Ca(2+) ions into excitable cells but are also involved in a variety of Ca(2+)-dependent processes, including muscle contraction, hormone or neurotransmitter release, and gene expression. Diriong et al. (1995) noted that calcium channels are multisubunit complexes and that the channel activity is directed by a pore-forming alpha-1 subunit, which is often sufficient to generate voltage-sensitive Ca(2+) channel activity. There are at least 6 classes of alpha-1 subunits: alpha-1A, B, C, D, E, and S, which are derived from 6 genes representing members of a gene family. The auxiliary subunits beta (e.g., 114207), alpha-2/delta, and gamma (e.g., 114209) regulate channel activity. In addition to full-length CACNA1A, use of an internal ribosomal entry site in the CACNA1A transcript generates the CACNA1A C-terminal polypeptide, or alpha-1ACT, which functions as a transcription factor that mediates cerebellar development (Du et al., 2013).\",\n",
        "    \"ATXN7 is a transcription factor that appears to be critically important for chromatin remodeling at the level of histone acetylation and deubiquitination. It is a core component of 2 different transcription coactivator complexes: the SPT3 (SUPT3H; 602947)/TAF9 (600822)/GCN5 (KAT2A; 602301) acetyltransferase (STAGA) complex, which has histone acetyltransferase activity, and the USP22 (612116) deubiquitination complex (summary by Sopher et al., 2011).\",\n",
        "    \"Spinocerebellar ataxia-8 (SCA8; 608768) is a neurodegenerative disorder caused by a CTG/CAG trinucleotide repeat expansion on chromosome 13q21 (see 603680.0001 and 613289.0001). Two genes span the CTG/CAG repeat and are expressed in opposite directions: ATXN8 (613289), which encodes a nearly pure polyglutamine expansion protein in the CAG direction, and ATXN8OS, which, when transcribed, produces a noncoding CUG expansion RNA (Moseley et al., 2006).\",\n",
        "    \"By positional cloning in the spinocerebellar ataxia-10 (SCA10; 603516) candidate region on chromosome 22q13-qter, Matsuura et al. (2000) identified a gene, ataxin-10 (ATXN10), encoding a deduced 475-amino acid protein with 82% identity with a presumed mouse ortholog (E46). By Northern blot analysis, Wakamiya et al. (2006) identified a 2-kb mRNA ATXN10 transcript.\",\n",
        "    \"The PPP2R2B gene encodes a brain-specific regulatory subunit B of protein phosphatase 2. Protein phosphatase 2A (PP2A), a heterotrimeric serine/threonine phosphatase, has been implicated in a variety of regulatory processes, including cell growth and division, muscle contraction, and gene transcription. PP2A is composed of a 36-kD catalytic subunit (176915), a highly homologous 65-kD structural subunit (176915), and any of several different regulatory subunits that control its specificity, including PPP2R2B (Mayer et al., 1991).\",\n",
        "    \"The RNA polymerase II transcription factor D (TFIID; see 313650) is a multisubunit complex essential for the expression of most, if not all, protein-encoding genes. The DNA-binding subunit of TFIID is the TATA box-binding protein (TBP).\",\n",
        "    \"The FGF14 gene encodes fibroblast growth factor-14, which is highly expressed in the brain, particularly in Purkinje cells where it interacts with voltage-gated channels to regulate neuronal excitability. FGF14 also plays a role in synaptic plasticity and neurogenesis in the hippocampus (summary by Piarroux et al., 2020). FGF14 belongs to a subclass of fibroblast growth factors that are expressed in the developing and adult central nervous system (Smallwood et al., 1996). For a discussion on the FHF gene family, see FGF12 (FHF1; 601513).\",\n",
        "    \"Frataxin is a nuclear-encoded mitochondrial iron chaperone involved in iron-sulfur biogenesis and heme biosynthesis. Some studies have also suggested that frataxin functions as an iron storage molecule, an antioxidant, and a tumor suppressor (summary by Schmucker et al. (2008)).\",\n",
        "    \"The RFC1 gene encodes the large subunit of replication factor C, a 5-subunit DNA polymerase accessory protein required for the coordinated synthesis of both DNA strands during replication or after DNA damage (summary by Cortese et al., 2019). It is a DNA-dependent ATPase that binds in a structure-specific manner to the 3-prime end of a primer hybridized to a template DNA, an activity thought intrinsic to the 140-kD component of this multisubunit complex (Bunz et al., 1993).\",\n",
        "    \"By Southern blot analysis of a 900-kb critical region on chromosome 16q21-q22, followed by sequencing analysis, Sato et al. (2009) identified 2.5- to 3.8-kb insertions containing pentanucleotide repeats including a (TGGAA)n sequence in all 160 affected individuals from 98 families with SCA31 (117210). PCR amplification followed by sequencing showed that the insertion consisted of a preceding TCAC sequence, and 3 pentanucleotide repeat components (TGGAA)n, (TAGAA)n, and (TAAAA)n in all patients tested. In a homozygous patient from whom the 900-kb critical region was derived, the authors found a (TGGAA)n sequence of greater than 110 repeats and a (TAAAATAGAA)n sequence of greater 112 repeats, both of which were too long to be read through. The (TGGAA)n and (TAAAATAGAA)n sequences were separated by a bridging sequence and (TAGAA)46. The insertions were located in introns of the BEAN (612051) and TK2 (188250) genes, which are on opposite strands and transcribed in opposite directions. These insertions were not identified in 99.77% of 800 Japanese and 60 American chromosomes, or in individuals with SCA4 (600223). However, 2 (0.23%) of 860 control chromosomes did carry similar smaller 1.5- or 2.0-kb insertions without (TGGAA)n sequences. Sato et al. (2009) concluded that the insertions in SCA31 patients exerted their toxicity either because of their length or the (TGGAA)n sequence, or because of both. The length of the SCA31 insertion was inversely correlated with the age at disease onset. Further analysis showed that the insertion site was identical for all insertions and was located at an Alu sequence. A single-nucleotide change in an intron of the TK2 gene segregated with SCA31 but was not considered to be pathogenic. The repeat insertions did not appear to cause splicing abnormalities or alterations in the expression levels of BEAN, TK2, or other nearby genes. Sato et al. (2009) demonstrated that the insertion transcribed in the direction of BEAN formed RNA foci in approximately 30 to 50% of Purkinje cell nuclei from SCA31 patients but not in cells from controls. RNA foci were not observed for antisense probes corresponding to TK2 transcripts in SCA31 or control brains. The splicing factors SFRS1 (600812) and SFRS9 (601943) were found to directly bind to (UGGAA)n, the transcribed sequence of (TGGAA)n, in vitro. In silico analysis showed that (TGGAA)n was abundant in centromeres of several human chromosomes, suggesting a role in heterochromatin or chromosomal structure.\",\n",
        "    \"The selective RNA-binding protein FMRP forms a messenger ribonucleoprotein complex that associates with polyribosomes, suggesting that it is involved in translation (Jin et al., 2004).\",\n",
        "    \"NOP56 is a component of the box C/D small nucleolar ribonucleoprotein complexes that direct 2-prime-O-methylation of pre-ribosomal RNA (rRNA) during its maturation. NOP56 is predicted to function in an early to middle step in pre-rRNA processing (Hayano et al., 2003).\",\n",
        "    \"The reelin (RELN; 600514) signaling pathway plays a critical role in the correct positioning of neurons within the developing brain. Animal studies have shown that DAB1 serves as an intracellular adaptor that is tyrosine phosphorylated when reelin binds to the lipoprotein receptors VLDLR (192977) and APOER2 (LRP8; 602600) on the surface of neurons (Huang et al., 2005).\"\n",
        "]\n",
        "\n",
        "\n",
        "result =(\n",
        "    df[\n",
        "    df[\"TRID\"].isin(SCA_RED) &\n",
        "    df[\"status\"].isin([\"PATHOGENIC\", \"INTERMEDIATE\"])\n",
        "    ]\n",
        "    .set_index(\"sample\")\n",
        "    .sort_index()\n",
        ")\n",
        "\n",
        "exclude_list = [\"SCA31_BEAN1\"]\n",
        "result = result[~result[\"TRID\"].isin(exclude_list)]\n"
      ],
      "metadata": {
        "id": "itqY1o7G5vSS"
      },
      "execution_count": 4,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "### sample_total\n",
        "\n",
        "sample_list=(\n",
        "    PC\n",
        "    .filter(like=\":AL\")\n",
        "    .columns\n",
        "    .str.replace(':AL', '', regex=False)\n",
        "    )\n",
        "\n",
        "motifs=(\n",
        "    PC\n",
        "    .loc[SCA_RED, 'MOTIFS']\n",
        "    .str.split(\",\", expand=True)\n",
        ")\n",
        "\n",
        "motifs.columns=[f\"motif_{i+1}\" for i in range(motifs.shape[1])]\n",
        "\n",
        "sample_AL={}\n",
        "sample_MC={}\n",
        "sample_total={}\n",
        "for sample in sample_list:\n",
        "    sample_AL[f\"{sample}_AL\"]=PC.loc[SCA_RED, f'{sample}:AL'].str.split(\",\", expand=True)\n",
        "    sample_AL[f\"{sample}_AL\"].columns=[f\"{sample}_AL_allele_{i+1}\" for i in range(sample_AL[f\"{sample}_AL\"].shape[1])]\n",
        "\n",
        "    sample_MC[f\"{sample}_MC\"]=PC.loc[SCA_RED, f'{sample}:MC'].str.split(\",\", expand=True)\n",
        "    sample_MC[f\"{sample}_MC\"].columns=[f\"{sample}_MC_allele_{i+1}\" for i in range(sample_MC[f\"{sample}_MC\"].shape[1])]\n",
        "    sample_MC_1=sample_MC[f\"{sample}_MC\"].iloc[:, 0].str.split(\"_\", expand=True)\n",
        "    sample_MC_1.columns=[f\"{sample}_MC_allele_1_motif_{i+1}\" for i in range(sample_MC_1.shape[1])]\n",
        "    sample_MC_2=sample_MC[f\"{sample}_MC\"].iloc[:, 1].str.split(\"_\", expand=True)\n",
        "    sample_MC_2.columns=[f\"{sample}_MC_allele_2_motif_{i+1}\" for i in range(sample_MC_2.shape[1])]\n",
        "\n",
        "    df_list=[motifs, sample_AL[f\"{sample}_AL\"], sample_MC_1, sample_MC_2]\n",
        "    sample_total[f\"{sample}\"]=pd.concat(df_list, keys=[\"motifs\", \"AL\", \"MC_allele_1\", \"MC_allele_2\"], axis=1)\n",
        "    sample_total[f\"{sample}\"].to_excel(f\"{sample}.xlsx\")\n",
        "\n",
        "pd.concat(sample_total, axis=1).to_excel(\"PC_total.xlsx\")"
      ],
      "metadata": {
        "id": "ikvpVrwE9fcR"
      },
      "execution_count": 5,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "### AL scatter plot\n",
        "sample_total_key_list=list(sample_total.keys())\n",
        "\n",
        "fig, axes = plt.subplots(len(sample_total_key_list), 2, figsize=(15,60))\n",
        "axes = axes.flatten() # Flatten the 2D array of axes for easy iteration\n",
        "\n",
        "def scatter_MC(ax, data, column, title):\n",
        "  data.plot.scatter(ax=ax, rot=45,\n",
        "                    x='TRID', y=column,\n",
        "                    xlabel='TRID', ylabel='AL',\n",
        "                    ylim=(0,1000),\n",
        "                    title=title)\n",
        "\n",
        "for i, key in enumerate(sample_total_key_list):\n",
        "  df = sample_total[key].AL.fillna(0).astype(int)\n",
        "  df.reset_index(inplace=True)\n",
        "\n",
        "  scatter_MC(axes[i*2], df, df.columns[1], f'{key}_allele_1')\n",
        "  scatter_MC(axes[i*2+1], df, df.columns[2], f'{key}_allele_2')\n",
        "\n",
        "plt.tight_layout()\n",
        "plt.show()"
      ],
      "metadata": {
        "id": "Y4jFGVXCqLuu"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "### practice\n",
        "\n",
        "sample"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 424
        },
        "id": "yMjW0_1g6qrb",
        "outputId": "b202ba8f-727c-41e2-de85-5b2c2c55935d"
      },
      "execution_count": 94,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                  TRID     sample   allele      TR      status  \\\n",
              "0           ADTKD_MUC1  B02_35311  allele1  2621.0  PATHOGENIC   \n",
              "1           ADTKD_MUC1  B02_35311  allele2  2621.0  PATHOGENIC   \n",
              "2           ADTKD_MUC1  B10_31534  allele1  2621.0  PATHOGENIC   \n",
              "3           ADTKD_MUC1  B10_31534  allele2  2621.0  PATHOGENIC   \n",
              "4           ADTKD_MUC1  B11_33267  allele1  2621.0  PATHOGENIC   \n",
              "...                ...        ...      ...     ...         ...   \n",
              "3061  pre-MIR7-2_CHNG3  SCA_15277  allele2    19.0  PATHOGENIC   \n",
              "3062  pre-MIR7-2_CHNG3  SCA_49841  allele1    19.0  PATHOGENIC   \n",
              "3063  pre-MIR7-2_CHNG3  SCA_49841  allele2    19.0  PATHOGENIC   \n",
              "3064  pre-MIR7-2_CHNG3  SCA_52664  allele1    19.0  PATHOGENIC   \n",
              "3065  pre-MIR7-2_CHNG3  SCA_52664  allele2    19.0  PATHOGENIC   \n",
              "\n",
              "              status_fmt  \n",
              "0     pathogenic(2621.0)  \n",
              "1     pathogenic(2621.0)  \n",
              "2       pathogenic(2621)  \n",
              "3       pathogenic(2621)  \n",
              "4       pathogenic(2621)  \n",
              "...                  ...  \n",
              "3061      pathogenic(19)  \n",
              "3062      pathogenic(19)  \n",
              "3063      pathogenic(19)  \n",
              "3064    pathogenic(19.0)  \n",
              "3065    pathogenic(19.0)  \n",
              "\n",
              "[3066 rows x 6 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-bb256220-7d42-4751-9188-f1d8b05f23dc\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>TRID</th>\n",
              "      <th>sample</th>\n",
              "      <th>allele</th>\n",
              "      <th>TR</th>\n",
              "      <th>status</th>\n",
              "      <th>status_fmt</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>ADTKD_MUC1</td>\n",
              "      <td>B02_35311</td>\n",
              "      <td>allele1</td>\n",
              "      <td>2621.0</td>\n",
              "      <td>PATHOGENIC</td>\n",
              "      <td>pathogenic(2621.0)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>ADTKD_MUC1</td>\n",
              "      <td>B02_35311</td>\n",
              "      <td>allele2</td>\n",
              "      <td>2621.0</td>\n",
              "      <td>PATHOGENIC</td>\n",
              "      <td>pathogenic(2621.0)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>ADTKD_MUC1</td>\n",
              "      <td>B10_31534</td>\n",
              "      <td>allele1</td>\n",
              "      <td>2621.0</td>\n",
              "      <td>PATHOGENIC</td>\n",
              "      <td>pathogenic(2621)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>ADTKD_MUC1</td>\n",
              "      <td>B10_31534</td>\n",
              "      <td>allele2</td>\n",
              "      <td>2621.0</td>\n",
              "      <td>PATHOGENIC</td>\n",
              "      <td>pathogenic(2621)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>ADTKD_MUC1</td>\n",
              "      <td>B11_33267</td>\n",
              "      <td>allele1</td>\n",
              "      <td>2621.0</td>\n",
              "      <td>PATHOGENIC</td>\n",
              "      <td>pathogenic(2621)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3061</th>\n",
              "      <td>pre-MIR7-2_CHNG3</td>\n",
              "      <td>SCA_15277</td>\n",
              "      <td>allele2</td>\n",
              "      <td>19.0</td>\n",
              "      <td>PATHOGENIC</td>\n",
              "      <td>pathogenic(19)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3062</th>\n",
              "      <td>pre-MIR7-2_CHNG3</td>\n",
              "      <td>SCA_49841</td>\n",
              "      <td>allele1</td>\n",
              "      <td>19.0</td>\n",
              "      <td>PATHOGENIC</td>\n",
              "      <td>pathogenic(19)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3063</th>\n",
              "      <td>pre-MIR7-2_CHNG3</td>\n",
              "      <td>SCA_49841</td>\n",
              "      <td>allele2</td>\n",
              "      <td>19.0</td>\n",
              "      <td>PATHOGENIC</td>\n",
              "      <td>pathogenic(19)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3064</th>\n",
              "      <td>pre-MIR7-2_CHNG3</td>\n",
              "      <td>SCA_52664</td>\n",
              "      <td>allele1</td>\n",
              "      <td>19.0</td>\n",
              "      <td>PATHOGENIC</td>\n",
              "      <td>pathogenic(19.0)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3065</th>\n",
              "      <td>pre-MIR7-2_CHNG3</td>\n",
              "      <td>SCA_52664</td>\n",
              "      <td>allele2</td>\n",
              "      <td>19.0</td>\n",
              "      <td>PATHOGENIC</td>\n",
              "      <td>pathogenic(19.0)</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3066 rows Ã— 6 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-bb256220-7d42-4751-9188-f1d8b05f23dc')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-bb256220-7d42-4751-9188-f1d8b05f23dc button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-bb256220-7d42-4751-9188-f1d8b05f23dc');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-927827a1-b690-4066-8687-b80bae5622cd\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-927827a1-b690-4066-8687-b80bae5622cd')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-927827a1-b690-4066-8687-b80bae5622cd button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "  <div id=\"id_e68aeab2-7b5c-4e48-adb2-2a4abc23273f\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('STATUS')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_e68aeab2-7b5c-4e48-adb2-2a4abc23273f button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('STATUS');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "STATUS",
              "summary": "{\n  \"name\": \"STATUS\",\n  \"rows\": 3066,\n  \"fields\": [\n    {\n      \"column\": \"TRID\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 73,\n        \"samples\": [\n          \"CCD_RUNX2\",\n          \"SCA6_CACNA1A\",\n          \"FAME3_MARCHF6\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"sample\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 21,\n        \"samples\": [\n          \"B02_35311\",\n          \"SCA9_70670213\",\n          \"SCA6_23844335\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"allele\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"allele2\",\n          \"allele1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"TR\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 341.47651608399184,\n        \"min\": 13.0,\n        \"max\": 5909.0,\n        \"num_unique_values\": 194,\n        \"samples\": [\n          155.0,\n          2617.0\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"status\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"UNKNOWN\",\n          \"INVALID\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"status_fmt\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 390,\n        \"samples\": [\n          \"pathogenic(2561)\",\n          \"intermediate(388)\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 94
        }
      ]
    }
  ]
}